### DISCUSSION TITLE
Keep getting GSON error

### DISCUSSION THREAD

**Discussion by ahsanbagwan** [langchain4j/langchain4j#1362] [Q&A]
This is while running easy RAG example. Can someone suggest what could be this? I have set proxy in OpenApi model.

Caused by: com.google.gson.stream.MalformedJsonException: Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 1 path $

              at com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:1659)

              at com.google.gson.stream.JsonReader.checkLenient(JsonReader.java:1465)

              at com.google.gson.stream.JsonReader.doPeek(JsonReader.java:602)

              at com.google.gson.stream.JsonReader.peek(JsonReader.java:433)

              at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:378)

**Comment by ahsanbagwan** [comment]
14:15:08.502 [main] WARN dev.langchain4j.internal.RetryUtils -- Exception
was thrown on attempt 1 of 3

java.lang.RuntimeException: com.google.gson.stream.MalformedJsonException:
Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 1
path $

              at
dev.ai4j.openai4j.SyncRequestExecutor.execute(SyncRequestExecutor.java:31)

              at
dev.ai4j.openai4j.RequestExecutor.execute(RequestExecutor.java:59)

              at
dev.langchain4j.model.openai.OpenAiChatModel.lambda$generate$0(OpenAiChatModel.java:153)

              at
dev.langchain4j.internal.RetryUtils$RetryPolicy.withRetry(RetryUtils.java:192)

              at
dev.langchain4j.internal.RetryUtils.withRetry(RetryUtils.java:229)

              at
dev.langchain4j.model.openai.OpenAiChatModel.generate(OpenAiChatModel.java:153)

              at
dev.langchain4j.model.openai.OpenAiChatModel.generate(OpenAiChatModel.java:113)

              at
dev.langchain4j.model.chat.ChatLanguageModel.generate(ChatLanguageModel.java:39)

              at com.rag_demo.RagDemo.main(RagDemo.java:85)

Caused by: com.google.gson.stream.MalformedJsonException: Use
JsonReader.setLenient(true) to accept malformed JSON at line 1 column 1
path $

              at
com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:1659)

              at
com.google.gson.stream.JsonReader.checkLenient(JsonReader.java:1465)

              at
com.google.gson.stream.JsonReader.doPeek(JsonReader.java:602)

              at com.google.gson.stream.JsonReader.peek(JsonReader.java:433)

              at
com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:378)

              at
retrofit2.converter.gson.GsonResponseBodyConverter.convert(GsonResponseBodyConverter.java:40)

              at
retrofit2.converter.gson.GsonResponseBodyConverter.convert(GsonResponseBodyConverter.java:27)

              at retrofit2.OkHttpCall.parseResponse(OkHttpCall.java:243)

              at retrofit2.OkHttpCall.execute(OkHttpCall.java:204)

              at
dev.ai4j.openai4j.SyncRequestExecutor.execute(SyncRequestExecutor.java:23)

              ... 8 common frames omitted

14:15:11.402 [main] WARN dev.langchain4j.internal.RetryUtils -- Exception
was thrown on attempt 2 of 3

java.lang.RuntimeException: com.google.gson.stream.MalformedJsonException:
Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 1
path $

              at
dev.ai4j.openai4j.SyncRequestExecutor.execute(SyncRequestExecutor.java:31)

              at
dev.ai4j.openai4j.RequestExecutor.execute(RequestExecutor.java:59)

              at
dev.langchain4j.model.openai.OpenAiChatModel.lambda$generate$0(OpenAiChatModel.java:153)

              at
dev.langchain4j.internal.RetryUtils$RetryPolicy.withRetry(RetryUtils.java:192)

              at
dev.langchain4j.internal.RetryUtils.withRetry(RetryUtils.java:229)

              at
dev.langchain4j.model.openai.OpenAiChatModel.generate(OpenAiChatModel.java:153)

              at
dev.langchain4j.model.openai.OpenAiChatModel.generate(OpenAiChatModel.java:113)

              at
dev.langchain4j.model.chat.ChatLanguageModel.generate(ChatLanguageModel.java:39)

              at com.rag_demo.RagDemo.main(RagDemo.java:85)

Caused by: com.google.gson.stream.MalformedJsonException: Use
JsonReader.setLenient(true) to accept malformed JSON at line 1 column 1
path $

              at
com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:1659)

              at
com.google.gson.stream.JsonReader.checkLenient(JsonReader.java:1465)

              at
com.google.gson.stream.JsonReader.doPeek(JsonReader.java:602)

              at com.google.gson.stream.JsonReader.peek(JsonReader.java:433)

              at
com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:378)

              at
retrofit2.converter.gson.GsonResponseBodyConverter.convert(GsonResponseBodyConverter.java:40)

              at
retrofit2.converter.gson.GsonResponseBodyConverter.convert(GsonResponseBodyConverter.java:27)

              at retrofit2.OkHttpCall.parseResponse(OkHttpCall.java:243)

              at retrofit2.OkHttpCall.execute(OkHttpCall.java:204)

              at
dev.ai4j.openai4j.SyncRequestExecutor.execute(SyncRequestExecutor.java:23)

              ... 8 common frames omitted

Exception in thread "main" java.lang.RuntimeException:
java.lang.RuntimeException: com.google.gson.stream.MalformedJsonException:
Use JsonReader.setLenient(true) to accept malformed JSON at line 1 column 1
path $

              at
dev.langchain4j.internal.RetryUtils$RetryPolicy.withRetry(RetryUtils.java:195)

              at
dev.langchain4j.internal.RetryUtils.withRetry(RetryUtils.java:229)

              at
dev.langchain4j.model.openai.OpenAiChatModel.generate(OpenAiChatModel.java:153)

              at
dev.langchain4j.model.openai.OpenAiChatModel.generate(OpenAiChatModel.java:113)

              at
dev.langchain4j.model.chat.ChatLanguageModel.generate(ChatLanguageModel.java:39)

              at com.amdocs.rag_demo.RagDemo.main(RagDemo.java:85)

Caused by: java.lang.RuntimeException:
com.google.gson.stream.MalformedJsonException: Use
JsonReader.setLenient(true) to accept malformed JSON at line 1 column 1
path $

              at
dev.ai4j.openai4j.SyncRequestExecutor.execute(SyncRequestExecutor.java:31)

              at
dev.ai4j.openai4j.RequestExecutor.execute(RequestExecutor.java:59)

              at
dev.langchain4j.model.openai.OpenAiChatModel.lambda$generate$0(OpenAiChatModel.java:153)

              at
dev.langchain4j.internal.RetryUtils$RetryPolicy.withRetry(RetryUtils.java:192)

              ... 5 more

Caused by: com.google.gson.stream.MalformedJsonException: Use
JsonReader.setLenient(true) to accept malformed JSON at line 1 column 1
path $

              at
com.google.gson.stream.JsonReader.syntaxError(JsonReader.java:1659)

              at
com.google.gson.stream.JsonReader.checkLenient(JsonReader.java:1465)

              at
com.google.gson.stream.JsonReader.doPeek(JsonReader.java:602)

              at com.google.gson.stream.JsonReader.peek(JsonReader.java:433)

              at
com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:378)

              at
retrofit2.converter.gson.GsonResponseBodyConverter.convert(GsonResponseBodyConverter.java:40)

              at
retrofit2.converter.gson.GsonResponseBodyConverter.convert(GsonResponseBodyConverter.java:27)

              at retrofit2.OkHttpCall.parseResponse(OkHttpCall.java:243)

              at retrofit2.OkHttpCall.execute(OkHttpCall.java:204)

              at
dev.ai4j.openai4j.SyncRequestExecutor.execute(SyncRequestExecutor.java:23)

**Comment by mczcqqchen** [comment]
@langchain4j Has this problem been resolved and how should it be resolved

**Reply by mczcqqchen** [reply]
[error.txt](https://github.com/user-attachments/files/17094614/error.txt)
I am using version 0.31.0 and I hope you can solve the problem. Thank you

**Reply by mczcqqchen** [reply]
streamgChatLanguageModel. generate() can run, chatLanguageModel. generate() reports this error
